app/src/main/java/io/intrepid/contest/screens/entrysubmission/entryname/EntryNameActivity.java

```diff
@@ -59,28 +53,30 @@ public void onEntryNameSubmitted() {
     }
 
     @Override
-    public void setContestName(String contestName) {
-        contestantWelcomeTextView.setText(getResources().getString(R.string.contestant_welcome_message, contestName));
+    public void showWelcomeMessage() {
+        setActionBarTitle(getResources().getString(R.string.contestant_welcome_message));
     }
 
     @Override
-    public void showEntryNameButton() {
+    public void enableEntryNameButton() {
+        entryNameNextButton.setEnabled(true);
         entryNameNextButton.setVisibility(android.view.View.VISIBLE);
     }
 
     @Override
-    public void hideEntryNameButton() {
+    public void disableEntryNameButton() {
+        entryNameNextButton.setEnabled(false);
         entryNameNextButton.setVisibility(android.view.View.GONE);
     }
 
     @Override
-    public void enableEntryNameButton() {
-        entryNameNextButton.setEnabled(true);
+    public void showIcon() {
```

Comment:
Ok! This icon concept only appears in two screens (name your contest and name your submission), so for now I only called it "`entryNameIcon`" (`R.id.entry_name_icon`).
I don't want to be specific about the icon shape (paper) because it might change.
Let me know if `showEntryNameIcon` is clear or if I should come up with a better concept.